Remove "Rejected" OEP status, and indicate they should be left unmerged

sarina opened the pull request:
Reflect reality that rejected proposals are frequently left unmerged even though OEP-1 had previously called for it.

Remove the unused "Rejected" state and codify that you should just close the PR, and leave a note as to why.

Addresses #428

kdmccormick reviewed (commented):
Thanks Sarina. I think this is an improvement over the current docs and I would not oppose merging it as-is.

That said, I feel that every instance in our official processes where we say "you may do this, or you can do that" without clear reason for doing one or the other is introducing a point of unnecessary ambiguity for our contributors. Writing an OEP is already challenging since it requires so much nuance and revision and convincing and difficult decision-making to reach a solid final document.... I think we can help OEP authors by making the mechanics of the process to be as clear-cut as possible. So, I would suggest to remove the "Rejected" state entirely, and instruct that rejected OEPs be closed instead of merged.

sarina commented:
@kdmccormick thanks for the feedback. I agree and added that change and reworked the PR title & description. Removing "Rejected" was a bit of a sweeping change and I'm not certain I did it all properly so I'd love your eyes for a second round.

kdmccormick reviewed (approved):
💯

pdpinch reviewed (approved):
👍
